packages/cli/src/commands/set-admin.ts

```diff
@@ -19,46 +43,111 @@ const register: (program: any) => any = (program) => program
   .description(description)
   .option('-f, --force', 'bypass a manual check')
   .withNetworkOptions()
+  .withNonInteractiveOption()
   .action(action);
 
-async function action(contractFullNameOrAddress: string, newAdmin: string, options: any): Promise<void | never> {
-  const { network, txParams } = await ConfigVariablesInitializer.initNetworkConfiguration(options);
+async function action(contractReference: string, newAdmin: string, options: any): Promise<void | never> {
+  const { force, network: networkInOpts, interactive  } = options;
+  const networkOpts = await promptForNetwork(options, () => getCommandProps());
+
+  const { network, txParams } = await ConfigVariablesInitializer.initNetworkConfiguration({ ...options, ...networkOpts });
   if (!await hasToMigrateProject(network)) process.exit(0);
 
-  let proxyAddress;
-  let contractAlias;
-  let packageName;
+  const { proxyReference, newAdmin: pickedNewAdmin } = await promptForProxies(contractReference, newAdmin, network, options);
+  const parsedContractReference = parseContractReference(proxyReference);
 
-  if (!contractFullNameOrAddress) throw Error('You have to specify at least a new admin address.');
+  if (!parsedContractReference) throw Error('You have to specify at least a new admin address.');
 
-  // we assume if newAdmin is empty it was specified as first argument
-  if (!newAdmin) {
-    newAdmin = contractFullNameOrAddress;
-    contractFullNameOrAddress = '';
-  }
-
-  if (!options.force) {
+  // has to be a standalone question from interactivity
+  // because it is security related and can't be disabled with interactivity set to false
+  if (!force) {
     const answers = await inquirer.prompt([
       {
         name: 'address',
         type: 'string',
         message: 'Please double check your address and type the last 4 characters of the new admin address. If you provide a wrong address, you will lose control over your contracts.',
```

Comment:
I find this message (and how and when it's prompted) somewhat confusing. Maybe just shortening it to a `Please, provide the last 4 characters of the specified address:` or something like that.

Reply:
What exactly is confusing? This is important message. Mishandling `set-admin` command may result in a loss of millions of dollars and reputation. `Please, provide the last 4 characters of the specified address` doesn't put required significance on it.

Reply:
I think that prompt messages should be as simple (and short) as possible, but you are right, in this case it will be better to be more explicit about the consequences.  

What is confusing for me is that the prompt message is not finishing with a reference to what the user should answer. WDYT about rephrasing it to something like:
`Warning! If you provide a wrong address, you will lose control over your contracts. Please double check your address and type the last 4 characters of the new admin address:`